### PR TITLE
feat(aio): update aio-shell class when sidenav opens or closes

### PR DESCRIPTION
**Pull request by petebacondarwin**
**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by sjtrimble — APPROVED** [review]
Just what is needed, thanks!

**Comment by petebacondarwin** [comment]
I need to add tests - tomorrow

File: aio/src/app/app.component.spec.ts [hunk 1]
```diff
@@ -267,15 +268,34 @@ describe('AppComponent', () => {
...
+
+      sideNav.componentInstance.opened = false;
+      sideNav.triggerEventHandler('close', {});
+      fixture.detectChanges();
+      expect(host.properties['className']).toContain('sidenav-closed');
+
+      sideNav.componentInstance.opened = true;
+      sideNav.triggerEventHandler('open', {});
+      fixture.detectChanges();
+      expect(host.properties['className']).toContain('sidenav-open');
```
Comment (gkalpak) [inline aio/src/app/app.component.spec.ts]: Nit: By using `toContain()` (here and above) we don't verify that classes are actually removed. Maybe you could add a second `not.toContain()` assertion.

**Review by gkalpak — APPROVED** [review]
One nit. LGTM otherwise.

File: aio/src/app/app.component.spec.ts (hunk 1 above)
Comment (petebacondarwin) [inline aio/src/app/app.component.spec.ts]: I agree. I'll make a better test check

File: aio/src/app/app.component.spec.ts [hunk 2]
```diff
@@ -262,21 +263,57 @@ describe('AppComponent', () => {
...
+      sideNav.triggerEventHandler('open', {});
+      fixture.detectChanges();
+      checkHostClass('sidenav', 'open');
     });
+
+    function checkHostClass(type, value) {
+      const regex = new RegExp('(?:^|\\s)' + type + '-(\\S*)', 'g');
+      const classes = host.properties['className'];
+      const matches = regex.exec(classes);
+      expect(matches && matches[1]).toEqual(value, `"${classes}" should contain "${type}-${value}"`);
```
Comment (gkalpak) [inline aio/src/app/app.component.spec.ts]: Nit: The error message will be confusing if `classes` _does_ contain `type-value` but not as the first match (e.g. `type-othervalue type-value`.

Comment (petebacondarwin) [inline aio/src/app/app.component.spec.ts]: Good point

File: aio/src/app/app.component.spec.ts [hunk 3]
```diff
@@ -262,21 +263,56 @@ describe('AppComponent', () => {
...
+      sideNav.triggerEventHandler('open', {});
+      fixture.detectChanges();
+      checkHostClass('sidenav', 'open');
     });
+
+    function checkHostClass(type, value) {
+      const classes = host.properties['className'];
+      const count = classes.split(' ').filter(c => c.indexOf(`${type}-`) === 0).length;
+      expect(count).toBeLessThanOrEqual(1, `"${classes}" should have only one class matching ${type}-*`);
+      expect(classes).toContain(`${type}-${value}`);
```
Comment (gkalpak) [inline aio/src/app/app.component.spec.ts]: This will fail when looking for `type-value` in `class="type-other ktype-value"` :grin:

Comment (petebacondarwin) [inline aio/src/app/app.component.spec.ts]: 👿

**Pull request merged**